diff --git a/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m b/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
index a1ffe69..3d7ea6d 100644
--- a/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
+++ b/node_modules/react-native-maps/ios/AirGoogleMaps/AIRGoogleMap.m
@@ -188,7 +188,10 @@ - (void)insertReactSubview:(id<RCTComponent>)subview atIndex:(NSInteger)atIndex
       [self insertReactSubview:(UIView *)childSubviews[i] atIndex:atIndex];
     }
   }
-  [_reactSubviews insertObject:(UIView *)subview atIndex:(NSUInteger) atIndex];
+    if (subview != nil) {
+        NSUInteger safeIndex = MIN(atIndex, _reactSubviews.count);
+        [_reactSubviews insertObject:(UIView *)subview atIndex:(NSUInteger) safeIndex];
+    }
 }
 #pragma clang diagnostic pop
 
@@ -236,7 +239,9 @@ - (void)removeReactSubview:(id<RCTComponent>)subview {
       [self removeReactSubview:(UIView *)childSubviews[i]];
     }
   }
-  [_reactSubviews removeObject:(UIView *)subview];
+    if (subview != nil) {
+        [_reactSubviews removeObject:(UIView *)subview];
+    }
 }
 #pragma clang diagnostic pop
 
diff --git a/node_modules/react-native-maps/ios/AirMaps/AIRMap.m b/node_modules/react-native-maps/ios/AirMaps/AIRMap.m
index 07a4638..1eb5ccb 100644
--- a/node_modules/react-native-maps/ios/AirMaps/AIRMap.m
+++ b/node_modules/react-native-maps/ios/AirMaps/AIRMap.m
@@ -135,7 +135,11 @@ - (void)insertReactSubview:(id<RCTComponent>)subview atIndex:(NSInteger)atIndex
             [self insertReactSubview:(UIView *)childSubviews[i] atIndex:atIndex];
         }
     }        
-    [_reactSubviews insertObject:(UIView *)subview atIndex:(NSUInteger) atIndex];
+
+    if (subview != nil) {
+        NSUInteger safeIndex = MIN(atIndex, _reactSubviews.count);
+        [_reactSubviews insertObject:(UIView *)subview atIndex:(NSUInteger) safeIndex];
+    }
 }
 #pragma clang diagnostic pop
 
@@ -166,7 +170,10 @@ - (void)removeReactSubview:(id<RCTComponent>)subview {
             [self removeReactSubview:(UIView *)childSubviews[i]];
         }
     }
-    [_reactSubviews removeObject:(UIView *)subview];
+
+    if (subview != nil) {
+        [_reactSubviews removeObject:(UIView *)subview];
+    }
 }
 #pragma clang diagnostic pop
 
